src/index.js

```diff
@@ -15,27 +15,37 @@ chrome.browserAction.setPopup({ popup: 'settings.html' })
 
 const frameState = {
   connected: false,
-  availableChains: []
+  availableChains: [],
+  currentChain: ''
 }
 
 function setChains (chains) {
-  frameState.availableChains = chains.map(c => parseInt(c))
+  frameState.availableChains = chains
+  if (settingsPanel) settingsPanel.postMessage(frameState)
+}
+
+function setCurrentChain (chain) {
+  frameState.currentChain = chain
+  if (settingsPanel) settingsPanel.postMessage(frameState)
 }
 
 provider.on('connect', () => {
   frameState.connected = true
-
-  provider.request({ method: 'wallet_getChains' }).then(setChains)
+  if (settingsPanel) settingsPanel.postMessage(frameState)
+  provider.request({ method: 'wallet_getChainDetails' }).then(setChains).catch(() => {})
 })
 
 provider.on('disconnect', () => { frameState.connected = false })
-provider.on('chainsChanged', setChains)
+provider.on('chainsChanged', () => {
+  provider.request({ method: 'wallet_getChainDetails' }).then(setChains).catch(() => {})
```

Comment:
will we just not show the chains if this fails? might want to log a warning or something

Reply:
Right, if this fails the chains module won't render. It was designed to fail silently b/c this will never work until they have beta.19. Yeah, we could add a log or something though.